### ISSUE TITLE
Add functionality to execute a user defined teardown after each test

### ISSUE THREAD

**Issue by gianfra-t** [pendulum-chain/wasm-deploy#28]
As part of #26, the tokens used are now persistent in the standalone chain (or any chain used to test the contracts).

This introduces a different behavior and one must be aware when running tests that the tokens minted will persist after each run. 
For this reason, some tests were modified in #26 to add a burn function before the minting, to ensure that the token balance starts at 0. (see [this for example)](https://github.com/pendulum-chain/wasm-deploy/blob/27a80969e0f1e5253121ca07e1cf4d0c3ac3a046/nabla/test/backstop.ts#L135).

### Proposed Enhancement
It would be interesting to have an analogous to the `setUp` function called  `teardown` for example, that gets executed after each test suite runs no matter if the execution is successful or not.

**Comment by gianfra-t** [comment]
@pendulum-chain/product The purpose of this ticket is to make a bit easier the writing and readability of tests. As of now it is completely possible to perform the same behavior just using the `setUp` function.

**Comment by prayagd** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Pendulum-6408a0365cda59c57d39bee5/issues/pendulum-chain/wasm-deploy/28?planning-poker) with Zenhub @adelarja @b-yap @ebma @TorstenStueber

**Comment by prayagd** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Pendulum-6408a0365cda59c57d39bee5/issues/pendulum-chain/wasm-deploy/28?planning-poker) with Zenhub @gianfra-t

**Comment by prayagd** [comment]
@gianfra-t i assume this is nice to have right? If yes can i move this to icebox?

**Comment by gianfra-t** [comment]
Yes it is a nice to have for the developer, not at all something that will add some extra capabilities.

**Comment by TorstenStueber** [comment]
This should be quite doable and, as far as I understand, is more powerful than what forge provides.
